target/ppc/fpu_helper.c

```diff
@@ -2172,17 +2222,12 @@ void helper_##op(CPUPPCState *env, ppc_vsr_t *xt,                             \
         t.fld = tp##_muladd(s1->fld, s3->fld, s2->fld, maddflgs, &tstat);     \
         env->fp_status.float_exception_flags |= tstat.float_exception_flags;  \
                                                                               \
-        if (unlikely(tstat.float_exception_flags & float_flag_invalid)) {     \
-            float_invalid_op_madd(env, tstat.float_exception_flags,           \
-                                  sfprf, GETPC());                            \
-        }                                                                     \
```

Comment:
I'm thinking on the implications of this change. Like is it ok to let the instruction continue (to the next iteration) or an invalid should be treated ASAP? Would this also cause a problem below where `*xt = t` is now set **before** the invalid exception is checked? My testes seemed ok but I didn't test the situation where exceptions are enabled

Reply:
It may be cheaper if we exit the loop earlier on exceptions, but since this is an `unlikely` case, I wouldn't be too worried about it. OTOH, the description of these instructions says that "If a trap-enabled exception occurs in any element of the vector, no results are written to VSR[XT]," so I guess we should call `do_float_check_status` before we set `*xt`.